### PR TITLE
Fix dashboard task, annotation list sorting

### LINKED ISSUE #3147: User Task list ordering is off
see https://discuss.webknossos.org/t/index-tab-show-finished-tasks-only-does-not-show-the-latest-task/1088

this uses AnnotationDAO.findAllFor, might have to fix the query there? or is the sorting overwritten in the frontend?

### PR DESCRIPTION
**Pull request by daniel-wer**
This PR fixes the sorting of the dashboard task list, which was sorted ascending by default, but should be descending as we're dealing with timestamps. I've also added a default sorting for the explorative annotation list, because this has been annoying me for quite some time.
Unrelatedly, I found that whenever a node was deleted, a superfluous updateTree update actions was sent to the server (in addition to the correct update actions). This was caused by an error in the `updateTreePredicate` which I fix in this PR as well.

/cc @fm3 

### URL of deployed dev instance (used for testing):
- https://fixtablesorting.webknossos.xyz

### Steps to test:
- Create (over a period of at least two minutes to get different creation dates) and finish some tasks. The finished/unfinished task list should be sorted correctly (newest task first). Same for the explorative annotation list.

### Issues:
- fixes #3147

------
- [x] Updated [changelog](../blob/master/CHANGELOG.md#unreleased)
- [ ] ~Updated [migration guide](../blob/master/MIGRATIONS.md#unreleased) if applicable~
- [ ] ~Updated [documentation](../blob/master/docs) if applicable~
- [ ] ~Needs datastore update after deployment~
- [x] Ready for review

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -25,9 +25,10 @@ For upgrade instructions, please check the [migration guide](MIGRATIONS.md).
 
 ### Fixed
 
-- Fixed a bug where large volume downloads contained invalid data.zip archives [#3086](https://github.com/scalableminds/webknossos/pull/3086)
-- Fixed a bug where non-privileged users were wrongly allowed to pause/unpause projects [#3097](https://github.com/scalableminds/webknossos/pull/3097)
-- Fixed a regression bug which caused the initial data loading to fail sometimes [#3149](https://github.com/scalableminds/webknossos/pull/3149)
+- Fixed a bug where large volume downloads contained invalid data.zip archives. [#3086](https://github.com/scalableminds/webknossos/pull/3086)
```
Comment (daniel-wer) [inline CHANGELOG.md]: I added a period after some changelog entries here :)

File: app/assets/javascripts/oxalis/model/sagas/skeletontracing_saga.js [hunk 2]
```diff
@@ -202,11 +202,11 @@ function* diffEdges(
 
 function updateTreePredicate(prevTree: TreeType, tree: TreeType): boolean {
   return (
-    prevTree.branchPoints !== tree.branchPoints ||
+    !_.isEqual(prevTree.branchPoints, tree.branchPoints) ||
     prevTree.color !== tree.color ||
     prevTree.name !== tree.name ||
     !_.isEqual(prevTree.comments, tree.comments) ||
-    !_.isEqual(prevTree.timestamp, tree.timestamp) ||
+    prevTree.timestamp !== tree.timestamp ||
```
Comment (philippotto) [inline app/assets/javascripts/oxalis/model/sagas/skeletontracing_saga.js]: Why is this necessary? Isn't timestamp a primitive value?

File: app/assets/javascripts/oxalis/model/sagas/skeletontracing_saga.js [hunk 3]
```diff
@@ -202,11 +202,11 @@ function* diffEdges(
 
 function updateTreePredicate(prevTree: TreeType, tree: TreeType): boolean {
   return (
-    prevTree.branchPoints !== tree.branchPoints ||
+    !_.isEqual(prevTree.branchPoints, tree.branchPoints) ||
```
Comment (philippotto) [inline app/assets/javascripts/oxalis/model/sagas/skeletontracing_saga.js]: Why is this necessary? Why doesn't a referential comparison suffice? If the branchpoints of a tree didn't change, we shouldn't create a new instance?
